.github/workflows/scorecard.yml

```diff
@@ -0,0 +1,72 @@
+# This workflow uses actions that are not certified by GitHub. They are provided
+# by a third-party and are governed by separate terms of service, privacy
+# policy, and support documentation.
+
+name: Scorecard supply-chain security
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  schedule:
+    - cron: '29 9 * * 4'
```

Comment:
Hmm -- this means the scan isn't activated even in this PR, so we don't get to see results, right? Suggest to change temporarily to get such a report. Prior to merge it seems prudent to add it to the [weekly scan](https://github.com/open-quantum-safe/liboqs/blob/main/.github/workflows/weekly.yml).

Reply:
By default no, not in the PR, in part because the action publishes to the code scanning results for the 'definitive' code in main.

The serif output file is also generated.

The action is not run at all for PRs. I could also trigger on PR and just generate the serif which we can just read. That could be just for PR verification, or on every PR. I'll assume the latter, we can remove before merging if required.

As an example of running the action on a merge commit see issue or https://github.com/planetf1/liboqs/actions/runs/8063103943

Reply:
> The serif output file is also generated.

With the latest commit it now is, thanks!

> we can remove before merging if required.

That depends on what this tool finds: Does it have any relationship whatsoever to the code (within a PR) or only procedural issues in the github setup? If the latter, it doesn't make any sense to run it on PRs.

Reply:
I can update the PR to be executed on merge only, and to just generate the serif for now. I'll leave the option to a) upload to scanning results b) post to openssf commented out.

We hopefully can then merge the PR.

I'll leave the issue open to track that followup discussion. I think the desired end goal would be to uncomment the two additional steps, though a final decision that we don't may happen too!

Reply:
> I can update the PR to be executed on merge only

I still don't quite understand why this should run during a PR/merge process: This seems to check (only) the "procedural" github setup not exactly influenced by code. Right or wrong? If right, I'd suggest running this only during the weekly tests (as well as at release). If wrong, I'd be grateful for an example of (code) problems this tool can point out and that a contributor can then fix during the PR process. Running only at merge seems wrong (either too late or too often).

> We hopefully can then merge the PR.

I'd prefer merging this only when we don't have a "red status" right away, i.e., the issues this highlights got fixed: This surely should not stall the next release planned for early next week.

Reply:
By 'merge' I mean when code is merged into main, by the process of merging the PR.
The actual workflow trigger is on a push to main
Yes, these are basically procedural checks

The intent is simply to record the state of the 'current' code.

We could run weekly - though the resource taken by this check is very low in terms of github actions. 

Scorecard is an OpenSSF initiative. It's just a checklist to provide some assurance that the project is adopting good practice. In some ways it's like having a badge or certificate for what you've learnt. You already know it, but want to demonstrate that in a standard way. 

This change is absolutely not relevant for the release  next week. I think the PR should wait until after that in any case

The results (if we go with the change here) are only published to the serif file attached to the workflow, nowhere else. But if it makes sense to wait until the underlying issues are fixed, I could look at PRs for those, and leave this in draft.

Reply:
The checks appear to be running clean on the v4 scorecard (as coded).

Please refer to the sarif results generated by the action.

An upgrade to scorecard v5 can be done as a new PR.

I will end off this PR update with a change to move the scans to release, once approval is secured. Closing this conversation.